erpcgen/src/templates/c_coders.template

```diff
@@ -228,7 +244,14 @@ codec->readData({$info.name}, {$info.sizeTemp} * sizeof({$info.builtinTypeName})
 {# Encode sending data #}
 {% def encodeBuiltinType(info) -----------------  %}
 {% if info.builtinType == "kStringType" %}
-codec->writeString(strlen((const char*){$info.name}), (const char*){$info.name});
+{
+    uint32_t {$info.stringLocalName}_len = strlen((const char*){$info.name});
+
+{% if info.stringAllocSize != info.stringLocalName & "_len"  %}
+    erpc_assert({$info.stringLocalName}_len <= {$info.stringAllocSize});
```

Comment:
you moved one top and one removed :D
my point was to change
```template
{% if info.stringAllocSize != info.stringLocalName & "_len"  %}

    erpc_assert({$info.stringLocalName}_len <= {$info.stringAllocSize});

{% endif %}
```
to 

```template
{% if info.stringAllocSize != info.stringLocalName & "_len"  %}

    erpc_assert({$info.stringLocalName}_len <= {$info.stringAllocSize});
{% endif %}
//newline was moved here outside
```
or

```template
//newline was moved here outside
{% if info.stringAllocSize != info.stringLocalName & "_len"  %}
    erpc_assert({$info.stringLocalName}_len <= {$info.stringAllocSize});

{% endif %}
```

Reply:
You need think how it looks when condition is true and when is false.

Reply:
I thought you will want no line between the assert to the codec->writeString. Such as you asked me to remove the line between the other assert to the codec->writeBinary. I am probably missing something, anyway I done it.